src/main/java/io/cdap/plugin/snowflake/sink/batch/StructuredRecordToCSVRecordTransformer.java

```diff
@@ -39,11 +43,11 @@
 public class StructuredRecordToCSVRecordTransformer {
   private static final Logger LOG = LoggerFactory.getLogger(StructuredRecordToCSVRecordTransformer.class);
 
-  public CSVRecord transform(StructuredRecord record) throws IOException {
+  public CSVRecord transform(StructuredRecord record) {
     List<String> fieldNames = new ArrayList<>();
     List<String> values = new ArrayList<>();
 
-    for (Schema.Field field : record.getSchema().getFields()) {
+    for (Schema.Field field : Objects.requireNonNull(Objects.requireNonNull(record.getSchema()).getFields())) {
```

Comment:
please revert this change.

We don't need to do this check here, if NPE occurs let it propagate upwards.

if `schema` is `null`, it should be validated in `configurePipeline()` & `prepareRun()`.